int64 channel
---
bool area1
bool area2
bool area3
bool success
